### ISSUE TITLE
unify tableFind with getRecord and getColumn, adding findRecord and findColumn

### ISSUE THREAD

**Issue by adrian-thurston** [influxdata/flux#2730]
Recently there have been requests to make `tableFind` + `get{Record,Column}` more robust by allowing `tableFind` to fail to find a table. Suppose the stream is empty, or there is no control over which keys are present. As things are, `tableFind` will error the program if it does not find a table.

The approach to solving this has been to return `nil` from a `tableFind` that does not find a table. The `getColumn` and `getRecord` functions can then be modified to handle this and return empty things. This works well for solving the user's issues.

The outstanding issue is what the rest of the functions do with this `nil`. With the above change a variable that is assigned a `nil` from a `tableFind` will cause a panic if it is used anywhere else in a flux program.

Normally this brings us to the type system to solve. We declare a new kind of thing for `tableFind` results and using the type system we allow this thing to get only into `getColumn` and `getRecord`. We currently have no concept of a `Table` in the type system. We have arrays of records to represent streams, but this is not the same thing that is returned from `tableFind`.

This brings us to the question ... is it worth it to add a new kind of thing to our type system just to pass between `tableFind` and `get{Record,Column}`? If we plan to expand on the things we can do with tables then this makes sense. The design for new flux seems to include this concept of a `Table`, but that is far away on the timeline. With current flux, there is nothing else to do with a `Table`.

Proposing instead to add two functions `findColumn` and `findRecord` that combine the tableFind with the conversion to record or array. They would be identical to their `get` original counterparts, except that they accept streams and take the `key` parameter of `tableFind`. By performing the find themselves they avoid exposing the table type to the user, which eliminates the need to modify the set of types.

TLDR: Add `findRecord` and `findColumn` functions to supersede the `tableFind` / `get{Column,Record}` combination we currently have.  The goal is to avoid exposing the unique kind of thing that is passed between them (a table — not a stream) and thereby avoid the need to expand the type system to represent that thing.
